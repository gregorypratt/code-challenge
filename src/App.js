import React, { Component } from 'react';
import SearchResults from './SearchResults';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      query: '',
      data: {}
    };
  }

  storeResults(data) {
    this.setState({
      data: JSON.parse(data)
    });
  }

  handleError({ errors }) {
    this.setState({
      errors,
      data: null
    });
  }

  search(e) {
    const query = e.target.value;

    this.setState({
      query
    });

    window.theMovieDb.search.getMovie({ query }, (data) => this.storeResults(data), (data) => this.handleError(data));
  }

  render() {
    const { data } = this.state;
    console.log(data);

    return (
      <div className="o-container o-container--large u-text u-window-box-super">
        <header>
          <h1 className="c-heading u-large">Search The Movie Database</h1>
        </header>
        <p className="c-paragraph">
          <input
            className="c-field"
            placeholder="e.g Fight Club"
            type="text"
            value={this.state.query}
            onInput={(e) => this.search(e)}
            onChange={(e) => this.search(e)}
          />
        </p>
        <SearchResults {...data} />
      </div>
    );
  }
}

export default App;
