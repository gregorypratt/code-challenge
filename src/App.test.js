import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

describe('App', () => {
  it('renders heading and search box', () => {
    const component = mount(<App />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('changes state on input', () => {
    const component = mount(<App />);

    component.instance().handleInput({ target: { value: 'test movie' } });
    expect(component.state()).toEqual({ data: {}, query: 'test movie', searching: true });
  });

  it('changes state on input', () => {
    const component = mount(<App />);

    component.instance().handleInput({ target: { value: 'test movie' } });
    expect(component.state()).toEqual({ data: {}, query: 'test movie', searching: true });
  });

  it('store results updates state correctly', () => {
    const component = mount(<App />);

    component.instance().storeResults(
      JSON.stringify({
        results: []
      })
    );
    expect(component.state()).toEqual({ data: { results: [] }, query: '', searching: false });
  });

  it('handle error updates state correctly', () => {
    const component = mount(<App />);

    component.instance().handleError();

    expect(component.state()).toEqual({ data: null, error: true, query: '', searching: false });
  });
});
