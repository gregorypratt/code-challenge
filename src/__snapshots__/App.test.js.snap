// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders heading and search box 1`] = `
<App>
  <div
    className="o-container o-container--large u-text u-window-box-medium"
  >
    <header>
      <h1
        className="c-heading u-super"
      >
        Search The Movie Database
      </h1>
    </header>
    <p
      className="c-paragraph"
    >
      <input
        autoFocus={true}
        className="c-field"
        onInput={[Function]}
        placeholder="e.g Fight Club"
        type="text"
        value=""
      />
    </p>
    <SearchResults>
      <section>
        <div
          className="o-grid o-grid--wrap"
        />
      </section>
    </SearchResults>
  </div>
</App>
`;
